components/mediation-ui/mediators-ui/org.wso2.carbon.mediator.enrich.ui/src/main/resources/web/enrich-mediator/edit-mediator.jsp

```diff
@@ -89,11 +92,13 @@
     } else if (enrichMediator.getTargetType().equals("custom")) {
         targetType = CUSTOM;
         if (enrichMediator.getTargetExpression() != null) {
-            SynapseXPath targetXPath = enrichMediator.getTargetExpression();
-            NameSpacesRegistrar nameSpacesRegistrar = NameSpacesRegistrar.getInstance();
-            if (targetXPath != null) {
-                nameSpacesRegistrar.registerNameSpaces(targetXPath, "mediator.enrich.target.val_ex", session);
-                targetValue = targetXPath.toString();
+            SynapsePath synapsePath = enrichMediator.getTargetExpression();
+            if (synapsePath != null) {
+                if (synapsePath instanceof SynapseXPath) {
+                    NameSpacesRegistrar nameSpacesRegistrar = NameSpacesRegistrar.getInstance();
+                    nameSpacesRegistrar.registerNameSpaces(synapsePath, "mediator.enrich.target.val_ex", session);
+                }
+                targetValue = synapsePath.toString();
```

Comment:
Duplicate code in 65-70.

Reply:
It's not recommended to write methods inside JSP pages.

Reply:
+1 fixed